src/Cms/PluginAssets.php

```diff
@@ -31,21 +31,34 @@ public static function clean(string $pluginName): void
 			$media  = $plugin->mediaRoot();
 			$assets = $plugin->assets();
 
-			// get outdated media files by comparing all
-			// files in the media folder against the current set
-			// of asset paths
-			$files  = Dir::index($media, true);
+			// get all media files
+			$files = Dir::index($media, true);
+
+			// get all active assets' paths from the plugin
 			$active = $assets->values(
-				fn ($asset) => $asset->mediaHash() . '/' . $asset->path()
+				function ($asset) {
+					$path  = $asset->mediaHash() . '/' . $asset->path();
+					$paths = [];
+					$parts = explode('/', $path);
+
+					// collect all path segments
+					// (e.g. foo/, foo/bar/, foo/bar/baz.css) for the asset
+					for ($i=1, $max = count($parts); $i <= $max; $i++) {
+						$paths[] = implode('/', array_slice($parts, 0, $i));
+
+						// TODO: remove when media hash is enforced as mandatory
+						$paths[] = implode('/', array_slice($parts, 1, $i));
```

Comment:
I didn't understand what exactly do this line.

**With this line:**
````php
  array(6) {
    [0]=>
    string(21) "1362650116-1646303478"
    [1]=>
    string(2) "js"
    [2]=>
    string(24) "1362650116-1646303478/js"
    [3]=>
    string(16) "js/jquery.min.js"
    [4]=>
    string(38) "1362650116-1646303478/js/jquery.min.js"
    [5]=>
    string(16) "js/jquery.min.js"
  }
````

**Without this line:**
Following seems more correctly than first one. Not sure.

````php
  array(3) {
    [0]=>
    string(21) "1362650116-1646303478"
    [1]=>
    string(24) "1362650116-1646303478/js"
    [2]=>
    string(38) "1362650116-1646303478/js/jquery.min.js"
  }
  ````

Reply:
I guess I understand. There is a case where it can be used without hash, so we need hashless paths as well. Right?

Reply:
For backwards compatibility we also support media paths that don't include the media hash. So if we have the asset path `js/jquery.min.js` with media hash `1362650116-1646303478` there are actually two locations in the media folder where it could end up:
- `1362650116-1646303478/js/jquery.min.js`
- `js/jquery.min.js`

This is why we're collecting path segments for both options:
- `1362650116-1646303478/`, `1362650116-1646303478/js/` and `1362650116-1646303478/js/jquery.min.js`
- `js/` and `js/jquery.min.js`

Reply:
Thank you! 🙌